troposphere/static/js/components/images/detail/availability/AvailabilityView.react.js

```diff
@@ -1,32 +1,56 @@
 import React from "react";
 import Backbone from "backbone";
 import stores from "stores";
-
+import Code from "components/common/ui/Code.react.js";
 
 export default React.createClass({
     displayName: "AvailabilityView",
 
     propTypes: {
-        version: React.PropTypes.instanceOf(Backbone.Model).isRequired
+        version: React.PropTypes
+            .instanceOf( Backbone.Model ).isRequired
     },
 
-    renderProviderMachine: function(provider_machine) {
+    renderProviderMachine( provider ) {
+        let { isSummary } = this.props;
+
         // Hide 'end-dated' provider_machines
-        let endDate = provider_machine.get("end_date");
+        let endDate = provider.get( "end_date" );
         if (endDate && endDate.isValid()) return;
 
+        // Assign strings for render
+        let machineID = provider.get( "uuid" );
+        let providerName = provider.get( "provider" ).name;
+
+        // Optionally render the UUID if not set as
+        // summary by parent
+        let optMachineID = isSummary ? null
```

Comment:
If you scroll up in the link a row you'll see the highlighted line, gh issue. Its a small point and maybe not worth bringing up, but the linked is what I prefer for ternarys. I'm basically commenting on your use of different placement of the `?` and `:`:

```
adfasd 
  ?  asdf : asdf

//or

asdfasdf ?
  { 
   asadf
  } : {
asdf 
}
```